linker/shared/src/test/scala/org/scalajs/linker/OptimizerTest.scala

```diff
@@ -0,0 +1,246 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker
+
+import scala.concurrent._
+
+import org.junit.Test
+import org.junit.Assert._
+
+import org.scalajs.ir.ClassKind
+import org.scalajs.ir.EntryPointsInfo
+import org.scalajs.ir.Definitions._
+import org.scalajs.ir.Trees._
+import org.scalajs.ir.Types._
+
+import org.scalajs.logging._
+
+import org.scalajs.junit.async._
+
+import org.scalajs.linker._
+import org.scalajs.linker.standard._
+
+import org.scalajs.linker.testutils._
+import org.scalajs.linker.testutils.IRAssertions._
+import org.scalajs.linker.testutils.TestIRBuilder._
+
+class OptimizerTest {
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  import OptimizerTest._
+
+  /** Generic code for the three methods below.
+   *
+   *  Check that `clone()` is never inlined when the result can be an array,
+   *  in several scenarios.
+   */
+  private def testCloneOnArrayNotInlinedGeneric(
```

Comment:
I'm unclear about what this method abstracts. If it is about mere duplication, why are the definitions of `reachClone`, `anArray`, `anObject` and `trivialCtor("LFoo")` not in here?

Reply:
Because I'm stupid ... I didn't realize that all those methods were completely duplicated. :sweat: 

Fixed.